### PR TITLE
adding udf functionality

### PR DESCRIPTION
**Pull request by shlee0605**
This pull request is for adding the UDF functionality for Peloton.

### CODE REVIEW

File: tests/udf/udf_benchmark.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        RETURN array1;
+    END IF;
+
+    l := array_upper(array2, 1);
+
+    SELECT array_agg(array1[i] + array2[i]) FROM generate_series(1, l) i INTO result;
+
+    RETURN result;
+END;
+$$ LANGUAGE plpgsql;
```
Comment (shlee0605) [inline tests/udf/udf_benchmark.sql]: @YangMann Let's place `*.sql` file in `scripts/testing/udf` and put c udf function implementations in `tests/udf`.

Comment (shlee0605) [inline tests/udf/udf_benchmark.sql]: @YangMann please put some queries that uses that function.

File: tests/udf/udf_benchmark.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
+CREATE TABLE udf_test_1 (col INT[]);
```
Comment (ZheyuanBu) [inline tests/udf/udf_benchmark.sql]: Did you test the script? Because I think peloton currently doesn't support array as an attribute.

**Comment by shlee0605** [comment]
merged to master

**Pull request closed without merging**